backend/src/agent/persistence/client/artifact/client.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2025 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package artifact provides client functionality for reading pipeline artifacts
+// using HTTP streaming.
+package artifact
```

Comment:
nit: can you rename this package to `artifact_client`, `artifact` feels unambiguous

Reply:
Done.